swift-extensions/UIPickerExtensions.swift

```diff
@@ -23,15 +23,15 @@ extension UIPickerView {
             guard let pickerViewModel = value else { return }
 
             self.delegate = self
-            observe(pickerViewModel.selectedElementIndex) { [weak self] (value: Int) in
-                self?.selectRow(value, inComponent: 0, animated: false)
-            }
-
             observe(pickerViewModel.elements) { [weak self] (value: [PickerItemViewModel]) in
                 self?.elements = value.map {
                     $0.displayName
                 }
             }
+
+            observe(pickerViewModel.selectedElementIndex) { [weak self] (value: Int) in
```

Comment:
Should we instead combine the two publishers in the same flow, so that when the elements change, the selecteElementIndex is taken into consideration (vise-versa)?

Reply:
With your change, I think that if the selectedElementIndex is dispatched before the elements (on another queue for example), we might still have the problem.

Reply:
Usually, the viewmodel is created long before it is binded to the view. But you may be right. There may be edge cases where there is a delay in loading the elements vs selecting the element index